File: 3312-number-of-changing-keys/3312-number-of-changing-keys.py
class Solution:
    def countKeyChanges(self, s: str) -> int:
        previous = ord(s[0])
        change = 0
        for i in range(1,len(s)):
            current = ord(s[i])
            if not (previous == current or abs(current - previous) == 32):
                change += 1
            previous = current
        return change

        

File: 3312-number-of-changing-keys/3312-number-of-changing-keys.go
import "math"

func countKeyChanges(s string) int {
    previous := int(s[0])
    change_count :=0
    var current int
    for i:=1;i<len(s);i++ {
        current = int(s[i])
        if previous != current && int(math.Abs(float64(previous- current))) != 32 {
            change_count += 1
        }
        previous = current
    }
    return change_count
}